src/pages/groupComparison/VennSimple.tsx

```diff
@@ -0,0 +1,230 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import {computed, observable} from "mobx";
+import autobind from "autobind-decorator";
+import {ComparisonGroup} from "./GroupComparisonUtils";
+import * as d3 from 'd3';
+import _ from "lodash";
+import LazyMemo from "../../shared/lib/LazyMemo";
+
+export interface IVennSimpleProps {
+    groups: {
+        uid: string;
+        cases: string[];
+    }[]; // either 2 or 3
+    x:number;
+    y:number;
+    width:number;
+    height:number;
+    uidToGroup: { [uid: string]: ComparisonGroup };
+    onClickRegion:(regionParams:{ intersectedUids:string[], excludedUids:string[] })=>void;
+    caseCountFilterName:string;
+}
+
+function blendColors(colors:string[]) {
+    switch (colors.length) {
+        case 1:
+            return colors[0];
+        case 2:
+            return d3.interpolateLab(colors[0], colors[1])(0.5);
+        case 3:
+        default:
+            return d3.interpolateLab(
+                d3.interpolateLab(colors[0], colors[1])(0.5),
+                colors[2]
+            )(0.5);
+    }
+}
+
+function regionMouseOver(e:any) {
+    e.target.setAttribute("fill", e.target.getAttribute("data-hover-fill"));
+}
+
+function regionMouseOut(e:any) {
+    e.target.setAttribute("fill", e.target.getAttribute("data-default-fill"));
+}
+
+function getExcludedIndexes(combination:number[], numGroupsTotal:number) {
+    const excl = [];
+    for (let i=0; i<numGroupsTotal; i++) {
```

Comment:
filter?

Reply:
would have to write a utility function to get the range [0,1,..., numGroupsTotal-1]. Nbd but just went this way instead.